[Zeroize] Breaking change for bounds

Consider the following code
```rust
use zeroize::{Zeroize, ZeroizeOnDrop};

#[derive(zeroize::ZeroizeOnDrop, zeroize::Zeroize)]
pub struct Test<A: Marker> {
    #[zeroize(skip)]
    field: Option<A>,
}

pub trait Secret: ZeroizeOnDrop + Zeroize{}

impl<A:Marker> Secret for Test<A>{

}

pub trait Marker {}
```

This code used to compile with:
```toml
[dependencies]
zeroize = { version = "1", features = ["derive"] }
zeroize_derive = { version = "=1.3.3" }
```

But now by updating the derive
```toml
[dependencies]
zeroize = { version = "1", features = ["derive"] }
zeroize_derive = { version = "=1.4.0" }
```

I get
```sh
error[E0277]: the trait bound `Test<A>: Zeroize` is not satisfied
  --> src/lib.rs:11:16
   |
11 | impl<A:Marker> Secret for Test<A>{
   |                ^^^^^^ the trait `Zeroize` is not implemented for `Test<A>`
   |
note: required by a bound in `Secret`
  --> src/lib.rs:9:35
   |
9  | pub trait Secret: ZeroizeOnDrop + Zeroize{}
   |                                   ^^^^^^^ required by this bound in `Secret`
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
11 | impl<A:Marker> Secret for Test<A> where Test<A>: Zeroize{
   |                                   ++++++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `test_zeroize` due to previous error
```

Which is due to https://github.com/RustCrypto/utils/pull/858


I bisected and downgrading derive to 81536ee278943c128c033813227c3bd57639e20a works perfectly with my example, upgrading to syn v2 broke something

--- a/zeroize/derive/src/lib.rs
+++ b/zeroize/derive/src/lib.rs
@@ -8,11 +8,10 @@ use proc_macro2::{Ident, TokenStream};
 use quote::{format_ident, quote};
 use syn::{
     parse::{Parse, ParseStream},
-    parse_quote,
     punctuated::Punctuated,
     token::Comma,
-    Attribute, Data, DeriveInput, Expr, ExprLit, Field, Fields, GenericParam, Lit, Meta, Result,
-    Variant, WherePredicate,
+    Attribute, Data, DeriveInput, Expr, ExprLit, Field, Fields, Lit, Meta, Result, Variant,
+    WherePredicate,
 };
 
 /// Name of zeroize-related attributes
@@ -39,16 +38,7 @@ fn derive_zeroize_impl(input: DeriveInput) -> TokenStream {
 
     let extra_bounds = match attributes.bound {
         Some(bounds) => bounds.0,
-        None => {
-            let mut out: Punctuated<WherePredicate, Comma> = Default::default();
-            for param in &input.generics.params {
-                if let GenericParam::Type(type_param) = param {
-                    let type_name = &type_param.ident;
-                    out.push(parse_quote! { #type_name: Zeroize })
-                }
-            }
-            out
-        }
+        None => Default::default(),
     };
 
     let mut generics = input.generics.clone();

--- a/zeroize/tests/zeroize_derive.rs
+++ b/zeroize/tests/zeroize_derive.rs
@@ -325,3 +325,29 @@ fn derive_zeroize_on_drop_generic() {
     #[derive(ZeroizeOnDrop)]
     struct Z<T: Zeroize>(Vec<T>);
 }
+
+#[test]
+fn derive_zeroize_unused_param() {
+    #[derive(Zeroize)]
+    struct Z<T> {
+        arr: [u32; 5],
+        #[zeroize(skip)]
+        skipped: T,
+    }
+}
+
+#[test]
+// Issue #878
+fn derive_zeroize_with_marker() {
+    #[derive(ZeroizeOnDrop, Zeroize)]
+    struct Test<A: Marker> {
+        #[zeroize(skip)]
+        field: Option<A>,
+    }
+
+    trait Secret: ZeroizeOnDrop + Zeroize {}
+
+    impl<A: Marker> Secret for Test<A> {}
+
+    trait Marker {}
+}

